src/game/common/system/datachunk.cpp

```diff
@@ -305,7 +306,6 @@ Utf16String DataChunkInput::Read_UnicodeString()
     }
 
     Decrement_Data_Left(sizeof(ch) * size);
-    string.Peek()[size] = L'\0';
```

Comment:
This is a bug and would crash the game if `size` happened to be zero, because no string would be allocated in `string` then. The code above does not match original code. I assume OmniBlade tried to implement it differently to accommodate charset better? Anyway, with the implementation of `string += ch` above a string null terminator is not necessary to be set, and original code also does not call `Peek` to begin with.